sdk/search/azure-search/src/main/java/com/azure/search/models/IndexingParameters.java

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+
+/**
+ * Represents parameters for indexer execution.
+ */
+@Fluent
+public final class IndexingParameters {
+    /*
+     * The number of items that are read from the data source and indexed as a
+     * single batch in order to improve performance. The default depends on the
+     * data source type.
+     */
+    @JsonProperty(value = "batchSize")
+    private Integer batchSize;
+
+    /*
+     * The maximum number of items that can fail indexing for indexer execution
+     * to still be considered successful. -1 means no limit. Default is 0.
+     */
+    @JsonProperty(value = "maxFailedItems")
+    private Integer maxFailedItems;
+
+    /*
+     * The maximum number of items in a single batch that can fail indexing for
+     * the batch to still be considered successful. -1 means no limit. Default
+     * is 0.
+     */
+    @JsonProperty(value = "maxFailedItemsPerBatch")
+    private Integer maxFailedItemsPerBatch;
+
+    /*
+     * Whether indexer will base64-encode all values that are inserted into key
+     * field of the target index. This is needed if keys can contain characters
+     * that are invalid in keys (such as dot '.'). Default is false.
+     */
+    @JsonProperty(value = "base64EncodeKeys")
+    private Boolean base64EncodeKeys;
+
+    /*
+     * A dictionary of indexer-specific configuration properties. Each name is
+     * the name of a specific property. Each value must be of a primitive type.
+     */
+    @JsonProperty(value = "configuration")
+    private Map<String, Object> configuration;
```

Comment:
Do we have a constant list of known supported configuration properties anywhere?  Might be something nice to include to make it easier to users to find them.

Reply:
@careyjmac In the .NET SDK, we had a custom class specifically to help with managing these configuration properties. I'm not sure if it exists in the Java SDK or not (I haven't looked for it).

Reply:
@brjohnstmsft @Careyjmac the Java SDK does not implemet this custom helper class. Added this to the backlog items for future version.

Reply:
In that case, it's not blocking this PR, but we should address it before we ship a preview. I'll find an owner for this.